### DISCUSSION TITLE
Working with modals

### DISCUSSION THREAD

**Discussion by ravmestre** [orchidsoftware/platform#1956] [Q&A]
Hi,

i'm trying to use a modal to add new records, and edit table records.

I have one page with the table, and in that page i have a ModalToggle in commandBar to New Records.

This part worked fine.

Now, i was trying to put an Edit button by table record, and open the modal with record data, but the modal opens always empty.

How can i achieve this? What am i missing? I am new to laravel and orchid platform. 

Thanks,
Ricardo Mestre

**Comment by gioacchinopoletto** [comment]
I have same issue.
I have tried loading aSync data, but form return always empty...

**Comment by gladiiss** [comment]
Try this
In your screen
`public function layout(): array
    {
        return [
            Layout::modal('createTableModal', [
                Layout::rows([]),
            ])->title('Create Table')
                ->size(Modal::SIZE_SM)
                ->applyButton('Create')
                ->closeButton('Cancel')
                ->rawClick(),
            Layout::modal('editTableModal', [
                Layout::rows([]),
            ])->title('Edit Table')
                ->size(Modal::SIZE_SM)
                ->applyButton('Save')
                ->closeButton('Cancel')
                ->rawClick()
                ->async('asyncGetTableData'),
        ];
    }`

Also add asyncGetTableData in same screen like this
`public function asyncGetTableData(Table $table): array
    {
        return [
            'table' => $table,
        ];
    }`

Don't forget to supply table_id as parameter
